imagebuildah/stage_executor.go

```diff
@@ -297,7 +297,27 @@ func (s *StageExecutor) digestSpecifiedContent(node *parser.Node) (string, error
 			}
 		}
 	}
+
+	// Make a map of our environment variables as a key and their
+	// value as the value in the map.  The values in argValues will
+	// look like "ENVVAR=myvalue".
+	envVarMap := make(map[string]string)
+	for _, arg := range argValues {
+		envVar := strings.Split(arg, "=")
+		if len(envVar) > 1 {
+			envVarMap[envVar[0]] = envVar[1]
+		}
+	}
+
 	for _, src := range srcs {
+		// If a src is an environment variable, try to resolve it.
+		if strings.HasPrefix(src, "$") {
+			src = src[1:]
+			if strings.HasPrefix(src, "{") && strings.HasSuffix(src, "}") {
+				src = src[1 : len(src)-1]
+			}
+			src = envVarMap[src]
```

Comment:
Should we report a failure whenr envVarMap[src] does not exists?

Reply:
Shouldn't happen, but probably not a bad idea.  I've bumped the tests up a bit more and wanted to play with ib.ProcessWord this morning, ran outta gas last night for both.  If the ib.ProcessWord doesn't have me rework this section, then I'll add an error.